pkg/api/kubernetesagentpool/types.go

```diff
@@ -0,0 +1,111 @@
+package kubernetesagentpool
+
+const (
+	APIVersion = "kubernetesagentpool"
```

Comment:
I though this is the internal/in memory API model just like github.com/Azure/acs-engine/pkg/api/types.go. Public/version API model is converted to internal model that ACS Engine then uses to generate the template.

Versioning and validation is done on the public API model (and the validation code is also called by the service).